utils.ml

```diff
@@ -55,3 +46,49 @@ module TimeHelper = struct
     | `Null -> Ok None
     | _ -> Error (`Msg "invalid json for Ptime.t option")
 end
+
+module Email = struct
+  type config = {
+    smtp_host : string;
+    smtp_port : int;
+    username : string;
+    password : string;
+  }
+
+  let validate_email email =
+    match Emile.of_string (Json.clean_string email) with
+    | Ok _ -> true
+    | Error s ->
+        Logs.err (fun m -> m "Emile-Email-Validation: %a" Emile.pp_error s);
+        false
+
+  let generate_signature uuid timestamp =
+    let timestamp_str = TimeHelper.string_of_ptime timestamp in
+    Base64.encode_string
+      (Cstruct.to_string
+         (Mirage_crypto.Hash.SHA256.digest
+            (Cstruct.of_string (uuid ^ "-" ^ timestamp_str))))
+
+  let generate_verification_link uuid timestamp =
+    let signature = generate_signature uuid timestamp in
+    let encoded_uuid = Base64.encode_string uuid in
+    "/auth/verify/token=" ^ encoded_uuid ^ "/" ^ signature
```

Comment:
the issue with this is: consider we know the user uuid (passed in here) -- and I'm at the moment not sure whether it is exposed (accidentally or as part of an API), we can do the entire token computation.

this means that a person who want to verify an email address they don't have access to, can do so - without receiving the email.